File: getting_started/bubble_sort/bubble_sort_python/bubble_sort.py
def bubbleSort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]


def printList(arr):
    for i in range(len(arr)):
        print(arr[i], end=" ")
    print()

if __name__ == "__main__":
    arr = [12, 11, 13, 5, 6, 7]
    print("Given array is ")
    printList(arr)

    bubbleSort(arr)

    print("Sorted array is")
    printList(arr)

File: getting_started/bubble_sort/bubble_sort_cpp/bubble_sort.cc
/**
 * Implementation of the bubble sort algorithm in C/C++
 * 
 * Author: Abdoulaye Diallo
 * 
 * Time Complexity: O(n²) 
*/


#include <iostream>
#include <stdlib.h>
#include <stdio.h>

void bubbleSort(int* arr, int n);
void printArray(int* arr, int n);

int main() {
    int arr[] = {12, 11, 13, 5, 6, 7};
    int n = sizeof(arr) / sizeof(arr[0]);
    printf("Given arrays is ");
    printArray(arr, n);

    bubbleSort(arr, n);

    printf("Sorted array is ");
    printArray(arr, n);
    return 0;
}

void bubbleSort(int* arr, int n) {
    for (int i=0; i < n; ++i)
    {
        for (int j =1 0; j < (n-i-1); ++j)
        {
            if (arr[j] > arr[j+1]) 
            {
                std::swap(arr[j], arr[j+1]);
            }
        }
    }
}

void printArray(int* arr, int n)
{
    for (int i=0; i < n; ++i)
        printf("%d ", arr[i]);
    printf("\n");
}
